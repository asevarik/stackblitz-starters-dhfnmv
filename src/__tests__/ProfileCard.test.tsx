import React from "react";
import { render } from "@testing-library/react";
import { ProfileCard } from "../Components";

describe("ProfileCard",()=>{
    const props = {
        username: 'JohnDoe',
        imgurl: 'john-doe.jpg',
        userId: '12345',
        type: 'User',
        score: 100,
      };
    const testValue = {imgUrl:"https://img.url",score:12,type:"user",userId:"123",username:"abc"}
    it("Should display the correct Test Values",()=>{
        const {getByTestId} = render(<ProfileCard imgurl={testValue.imgUrl} score={testValue.score} type={testValue.type} userId={testValue.userId} username={testValue.username}/>)
        const score = getByTestId("userScore").textContent;
        const uname = getByTestId("userName").textContent;
        const userId = getByTestId("userId").textContent;
        const userType = getByTestId("userType").textContent;
        expect(uname).toBe(testValue.username);
        expect(score).toBe(`score : ${testValue.score}`);
        expect(userId).not.toBeNull()
        expect(userType).toBe(`Type : ${testValue.type}`)
     })
     it('matches snapshot', () => {
        const { container } = render(<ProfileCard {...props} />);
    
        // Check if the rendered component matches the snapshot
        expect(container).toMatchSnapshot();
      });
})