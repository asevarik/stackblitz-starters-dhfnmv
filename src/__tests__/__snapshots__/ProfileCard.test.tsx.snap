// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileCard matches snapshot 1`] = `
<div>
  <div>
    <div
      class="card w-96 bg-base-200 shadow-xl"
    >
      <div
        class="card-body"
      >
        <h2
          class="card-title capitalize"
          data-testid="userName"
        >
          JohnDoe
        </h2>
        <section
          class="mt-6 flex gap-7 items-center justify-center"
        >
          <div
            class="avatar "
          >
            <div
              class="w-24 rounded-full bg-slate-50"
            >
              <img
                src="john-doe.jpg"
              />
            </div>
          </div>
          <div
            class="flex-1  min-h-[96px] flex flex-col gap-4 justify-start items-start"
          >
            <label
              class="capitalize"
              data-testid="userId"
            >
              Id : 
              12345
            </label>
            <label
              class="capitalize"
              data-testid="userType"
            >
              Type : 
              User
            </label>
            <label
              class="capitalize"
              data-testid="userScore"
            >
              score : 
              100
            </label>
          </div>
        </section>
      </div>
    </div>
  </div>
</div>
`;
